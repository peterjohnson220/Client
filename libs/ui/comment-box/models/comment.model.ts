export interface Comment {
  Content: string;
  CommentId?: number;
  FullName?: string;
  CompanyName?: string;
  ReplyCount?: number;
  Replies?: Comment[];
  CreateDate?: Date;
  ParentCommentId?: number;
}
