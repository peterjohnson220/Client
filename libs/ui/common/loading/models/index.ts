export * from './loading-progress-bar.model';
