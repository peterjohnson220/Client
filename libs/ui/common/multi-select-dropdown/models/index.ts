export * from './multi-select-item-group.model';
