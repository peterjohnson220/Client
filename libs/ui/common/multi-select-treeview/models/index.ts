export * from './treeview-item.model';
