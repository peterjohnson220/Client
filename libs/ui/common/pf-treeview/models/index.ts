export * from './treeview-mode.enum';
export * from './treeview-theme.enum';
