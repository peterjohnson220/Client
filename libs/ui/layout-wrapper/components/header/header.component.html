<nav class="navbar navbar-expand navbar-dark header-background-color">
  <a class="navbar-brand mr-auto">
    <img class="logo-style" [src]="payfactorsLogo" />
  </a>
  <ul class="navbar-nav">
    <li class="nav-item">
      <pf-layout-wrapper-user-menu></pf-layout-wrapper-user-menu>
    </li>
  </ul>
</nav>
