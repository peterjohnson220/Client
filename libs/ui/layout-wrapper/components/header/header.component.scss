@import 'client';

.navbar{
  background-color: $test;
}
