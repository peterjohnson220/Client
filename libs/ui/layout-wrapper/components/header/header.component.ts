import { Component, Input } from '@angular/core';

import { NavigationLink, UserContext, HomePageLink } from 'libs/models';

@Component({
  selector: 'pf-layout-wrapper-header',
  templateUrl: './header.component.html',
  styleUrls: ['./header.component.scss']
})
export class HeaderComponent {
  payfactorsLogo = './assets/payfactors-transparent.png';

  @Input() gettingDropDownNavigationLinks: boolean;
  @Input() gettingDropdownNavigationLinksError: boolean;

  @Input() dropdownNavigationLinks: NavigationLink[];
  @Input() userContext: UserContext;
  @Input() homePageLink: HomePageLink;

  constructor() {}
}
