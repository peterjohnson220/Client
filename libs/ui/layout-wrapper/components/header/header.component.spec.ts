import { Component, NO_ERRORS_SCHEMA } from '@angular/core';
import { TestBed, ComponentFixture } from '@angular/core/testing';

import { generateMockUserContext, generateMockNavigationLink, generateMocKHomePageLink} from 'libs/models';
import { AbstractFeatureFlagService } from 'libs/core';

import { HeaderComponent } from './header.component';

describe('User Menu', () => {
  let fixture: ComponentFixture<HeaderComponent>;
  let instance: HeaderComponent;

  // Configure Testing Module for before each test
  beforeEach(() => {
    TestBed.configureTestingModule({
      declarations: [
        HeaderComponent
      ],
      providers: [
        {
          provide: AbstractFeatureFlagService,
          useValue: { bindEnabled: jest.fn() },
        }
      ],
      // Shallow Testing
      schemas: [ NO_ERRORS_SCHEMA ]
    });

    fixture = TestBed.createComponent(HeaderComponent);
    instance = fixture.componentInstance;
  });

  it('should show the header', () => {

    instance.userContext = generateMockUserContext();
    instance.dropdownNavigationLinks = [];

    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('should show homepage link, when provided', () => {

    instance.userContext = generateMockUserContext();
    const dropDownLink1 = generateMockNavigationLink();

    instance.dropdownNavigationLinks = [dropDownLink1];
    instance.homePageLink = generateMocKHomePageLink();
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

});
