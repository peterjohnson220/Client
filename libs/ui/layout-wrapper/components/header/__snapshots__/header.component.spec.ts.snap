// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Menu should show homepage link, when provided 1`] = `
<pf-layout-wrapper-header
  dropdownNavigationLinks={[Function Array]}
  homePageLink={[Function Object]}
  userContext={[Function Object]}
>
  <nav
    class="navbar navbar-expand navbar-dark fixed-top"
  >
    
    <a
      class="navbar-brand mr-auto"
      href="url"
    >
      <img
        class="logo-style"
        src="Img Src"
      />
    </a>
    
    
    <pf-layout-wrapper-user-menu />
  </nav>
</pf-layout-wrapper-header>
`;

exports[`User Menu should show the header 1`] = `
<pf-layout-wrapper-header
  dropdownNavigationLinks={[Function Array]}
  userContext={[Function Object]}
>
  <nav
    class="navbar navbar-expand navbar-dark fixed-top"
  >
    
    <a
      class="navbar-brand mr-auto"
      href=""
    >
      <img
        class="logo-style"
        src="null"
      />
    </a>
    
    
    <pf-layout-wrapper-user-menu />
  </nav>
</pf-layout-wrapper-header>
`;
